ui/src/common/utils.js

```diff
@@ -48,6 +49,10 @@ export function getSizeOfArrayOrImmutableList(arrayOrList) {
   return Array.isArray(arrayOrList) ? arrayOrList.length : arrayOrList.size;
 }
 
+export function getFromObjectOrImmutableMap(objectOrMap, key) {
+  return Map.isMap(objectOrMap) ? objectOrMap.get(key) : objectOrMap[key];
```

Comment:
maybe an option https://github.com/rtfeldman/seamless-immutable we are adding more and more helpers like this

Reply:
As far as I can see this is completely different immutable library. Another thing is that we shouldn't add more of these but do a refactor to improve consistency at some point... Like all dumb components should take `JS` props and all state tree should be `immutable`.